### ISSUE TITLE
Python 3.8.7+ (3.9.1+) support (and Ubuntu 20.04)

### ISSUE THREAD

**Issue by segatrade** [googlecolab/colabtools#1422]
Bug report for Colab: http://colab.research.google.com/.

- Describe the current behavior:
It's impossible to use Python 3.8:
1) Some tools require Python >3.7. 
2) Python 3.8 is faster than 3.6.
3) Python 3.6 End of full support since 2018-12-24.
4) Ubuntu 20.04 LTS available with default Python 3.8 and will be main OS on servers for next 2 years.

- Describe the expected behavior:
Python 3.8 is available in Colab.

**Comment by colaboratory-team** [comment]
No ETA at this time, but tracked internally as b/112544213.

**Comment by datametrician** [comment]
+1

**Comment by segatrade** [comment]
this getting real problem:
```
numba.core.errors.UnsupportedError: Failed in nopython mode pipeline (step: analyzing bytecode)
'try' block not supported until python3.7 or later
```
numba is the one of the fastest solutions with python perfomance, especially with cuda. It's a pity that py 3.7/3.8 still unavailable in colab. There is any updates?

**Comment by johnnync13** [comment]
@colaboratory-team then, since it is a bit late to upgrade directly to python 3.9

**Comment by habanoz** [comment]
Python upgrade is needed. Please act.

**Comment by segatrade** [comment]
> @colaboratory-team then, since it is a bit late , you could to update directly to python 3.9

Python 3.9 possible not all library support it well. 3.8 is sweetspot or at least 3.7. But anyway 3.6 already obsolete

**Comment by johnnync13** [comment]
@segatrade yes, 3.9 not support all libraries butthe best thing would be to put it on and let everyone use the version they want.

**Comment by ogrisel** [comment]
scikit-learn 0.24.0 will be the last version to support Python 3.6. Supporting too many Python versions is expensive for Open Source maintainers and CI infrastructure. It would be great for colab to support a newer version of Python (ideally 3.9 but possibly 3.8 if all packages are not yet available for 3.9 at the time of the migration).

**Comment by craigcitro** [comment]
Just to say: we're working on this, though I can't promise a particular date.

**Comment by lbugnon** [comment]
Hi, given that you can actually install and run python3.x, why not to be able to change it in the main colab? I have packages that can run as a scritp, calling the right python version, but it is impossible to use them as an import on a cell

**Comment by segatrade** [comment]
I am stuck with updating on colab:
setup.py - Remove support for 3.6 - 2 months ago
:(

**Comment by n2cholas** [comment]
The latest stable NumPy release ([1.20.0](https://numpy.org/doc/stable/release/1.20.0-notes.html)) is only available for Python 3.7-3.9. I understand that this is being worked on, but hopefully the NumPy incompatibility provides justification to increase the priority of this issue.

**Comment by craigcitro** [comment]
🎉 🎉 🎉  Python 3.7 should be live for everyone as of today! 🎉 🎉 🎉 

Our goal is to keep the updates coming, but feel free to file issues when there are specific libraries and use-cases that are blocked on the lack of a new python version.

**Comment by Telofy** [comment]
Awesome, thank you! Is there a ticket for Python 3.8 or 3.9 support that I can subscibe to so I get notified when it’s supported too? :-D

**Comment by craigcitro** [comment]
For better or worse, the first place to be notified about changes is in Colab itself -- changes and features are often live for several days before we circle back and update pending issues. 

If you have a use-case or workflow that's blocked on a new python version, feel free to file a separate issue.

**Comment by pdxjohnny** [comment]
![image](https://user-images.githubusercontent.com/5950433/108926379-28fb6980-75f3-11eb-9f6c-43de1200a9c9.png)

Nice!

**Comment by johnnync13** [comment]
only 3.7? :(

**Comment by segatrade** [comment]
At the first moment I was upset too that only Python 3.7, but now I understand that's is right decision. Some libs still support 3.7 only in beta (https://www.pypy.org/compat.html). 

A lot of libraries use NEP 29 deprecation policy https://numpy.org/neps/nep-0029-deprecation_policy.html
On Dec 26, 2021 drop support for Python 3.7

https://github.com/pandas-dev/pandas/issues?q=Drop+Python+3.7+support

**Comment by colaboratory-team** [comment]
#1880 tracks the ubuntu 20.04 update.
